### ISSUE TITLE
Image Preprocessing 可能造成 Captioning 效果不佳

### ISSUE THREAD

**Issue by snsd0805** [hsuanchia/Image-caption#16]
這是我自己重新架 Image Captioning model 時候發現的問題。這是我的 [model 連結](https://colab.research.google.com/drive/12QI3IIaeVN7ONLSmVXPZMRGajOKHO4G9?usp=sharing)

model 基本上跟 GitHub 上的[這個](https://github.com/hsuanchia/Image-caption/blob/main/imgcap_0427.ipynb)相同，但是我發現我的模型 loss 可以降到 **0.1438**，sparce accuracy 是 **0.9622**

雖然說應該是 overfitting ，出來的句子也不太符合圖片，但是我認為做 predict 時圖片的處理應該要該產生 feature maps 的方法相同。當然這可以再討論看看。

我的圖片讀取方式是
```python
from tensorflow.keras.preprocessing import image
from tensorflow.keras.applications.vgg16 import preprocess_input

img = image.load_img('{}/{}'.format('/home/ujuone/Documents/coco/Picture/train2017', imgFile['file']), target_size=(224, 224))

x = image.img_to_array(img)
x = np.expand_dims(x, axis=0)
x = preprocess_input(x)

features = model.predict(x)
```

**Comment by hsuanchia** [comment]
我做predict跟產生feature map時的image preprocessing是一樣的，同樣都是先經過load_image的function，再升維，再做prediction，之後直接丟到decoder了。當然我現在的處理方法可能不好，可以改成你上方的這段code，但我認為主要的影響還是overfitting，先嘗試著把Dropout加進去和使用多一點的資料看能不能改善問題。你想想看，我們可以再討論討論。

**Comment by snsd0805** [comment]
我意思是說 你做predict的處理要跟training data的處理一樣，你現在training data的feature maps不是從我這邊拿的嗎？
我在想不一樣的是preprocess_input那個函式，我查過好像會經過標準化。

**Comment by hsuanchia** [comment]
好，那我把 [generate_caption](https://github.com/hsuanchia/Image-caption/blob/main/generate_caption.ipynb) 的 image preprocess 改成跟你上面的這個一樣好了，確實是應該要統一一下沒錯。
我再去多了解一下preprocess_input的運作好了。

**Comment by hsuanchia** [comment]
已將code補上 [generate_caption](https://github.com/hsuanchia/Image-caption/blob/main/generate_caption.ipynb) 裡面了
確實像你所說，改變處理image preprocessing的方法後，有影響到後面predict的結果，之後我training也會統一使用你上面這段code。

**Comment by snsd0805** [comment]
我先 closed
